src/framework/global/settings.cpp

```diff
@@ -110,6 +116,10 @@ Settings::Items Settings::readItems() const
         result[item.key] = item;
     }
 
+    if (multiInstancesProvider()) {
```

Comment:
BTW it might make sense to put IF_ASSERT here and below

Reply:
No, just ifs are needed here, and I think it's better to keep them. They are needed in tests, in tests, there is a call to the settings.